File: leetcode/58. Length of Last Word/main.ex
defmodule Solution do
  @spec length_of_last_word(s :: String.t) :: integer
  def length_of_last_word(s) do
    String.split(s, " ")
    |> Enum.filter(&(String.length(&1) > 0))
    |> Enum.at(-1)
    |> String.length()
  end
end

File: leetcode/58. Length of Last Word/main.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        // iterators in rust are such a pleasure to compose.
        s
            .split(' ')
            .rev()
            .filter(|&s| s.len() > 0)
            .next()
            .map_or(0, |s| s.len() as i32)
    }
}